projects/firmware_homework/src/main.c

```diff
@@ -0,0 +1,46 @@
+// Standard library includes
+#include <stdint.h>
+#include <stdio.h>
+// Midnight Sun includes
+#include "delay.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+// Macros
+#define DELAY_TIME 500
```

Comment:
Nitpick: we usually like to suffix our macros with the unit, so this should be `DELAY_TIME_MS`.